csharp/lib/src/lib.rs

```diff
@@ -26,37 +31,96 @@ pub struct Client {
     runtime: Runtime,
 }
 
-fn create_connection_request(
-    host: String,
-    port: u32,
-    use_tls: bool,
+/// Convert raw C string to a rust string.
+/// # Safety
+/// Unsafe function because creating a string from a pointer.
+unsafe fn ptr_to_str(ptr: *const c_char) -> &'static str {
```

Comment:
Why return a string slice? I think you should be able to return an owned String here, which would also mean you don't need the `into` call later.

Reply:
Changed to `String`, but it requires more `into()` though. Protobuf works with `Chars`, not with strings.

Reply:
I thought about this for a while and I think I still prefer the owned string being returned for now, even though it means extra `into` calls. We'll see if the AWS team likes this or not.